contracts/booking-token/CancellationUtils.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+/**
+ * @title CancellationUtils
+ * @notice Library for packing and unpacking cancellation-related data into a single
+ * uint256. Each field uses 16 bits, allowing values up to 65,535.
+ *
+ * Layout:
+ * [rejection_reason_version(16)][rejection_reason(16)][cancellation_reason_version(16)][cancellation_reason(16)]
+ * [------------------REJECTION_MASK------------------][-------------------CANCELLATION_MASK--------------------]
+ *
+ * The versions values are the package versions from the [Camino Messenger Protocol](https://github.com/chain4travel/camino-messenger-protocol)
+ *
+ * Cancellation Reason: camino-messenger-protocol/proto/cmp/services/cancellation/<version>/reason.proto
+ * Rejection Reason   : camino-messenger-protocol/proto/cmp/services/cancellation/<version>/rejection_reason.proto
+ */
+library CancellationUtils {
+    // Constants for bit manipulation
+    uint256 private constant SEGMENT_MASK = 0xFFFF; // 16 bits mask
+    uint256 private constant CANCELLATION_REASON_SHIFT = 0; // First 16 bits
+    uint256 private constant CANCELLATION_VERSION_SHIFT = 16; // Next 16 bits
+    uint256 private constant REJECTION_REASON_SHIFT = 32; // Next 16 bits
+    uint256 private constant REJECTION_VERSION_SHIFT = 48; // Final 16 bits
+
+    // Masks for different sections
+    uint256 private constant CANCELLATION_MASK = 0xFFFFFFFF;
+    uint256 private constant REJECTION_MASK = 0xFFFFFFFF00000000;
+
+    // Custom errors
+    error ValueExceedsMaxSize(uint256 value, uint256 maxSize);
+
+    /**
+     * @notice Packs cancellation and rejection data into a single uint256
+     * @param cancellationReason The reason for cancellation (max 16 bits)
+     * @param cancellationVersion Version of the cancellation reason enum (max 16 bits)
+     * @param rejectionReason The reason for rejection (max 16 bits)
+     * @param rejectionVersion Version of the rejection reason enum (max 16 bits)
+     * @return packed The packed uint256 containing all data
+     */
+    function packReasons(
+        uint256 cancellationReason,
+        uint256 cancellationVersion,
+        uint256 rejectionReason,
+        uint256 rejectionVersion
+    ) internal pure returns (uint256) {
+        // Check if any value exceeds 16 bits
+        if (cancellationReason > SEGMENT_MASK) {
```

Comment:
Is there a reason why the parameters are uint256 and checking the uint16 limits?
Why not use uint16 as a parameter and be sure that it can never exceed the limits?